extensions/pkg/webhook/certificates.go

```diff
@@ -124,6 +125,13 @@ func generateNewCAAndServerCert(mode, namespace, name, url string) (*secrets.Cer
 		ipAddresses []net.IP
 	)
 
+	serverName := url
+	serverNameData := strings.SplitN(url, ":", 3)
```

Comment:
Why not use `url.Parse` instead?

Reply:
Short answer: Because the string in question is not an URL.

Long answer: I haven't considered the `url.Parse` method before because even that the command line argument contains url in the name it does not contain a scheme or slashes, only the host and port portion of it. Looking at the [documentation](https://golang.org/pkg/net/url/#URL) the scheme is clearly required for `Parse()` to interpret the string given. I've now tested `url.Parse` without success, `urlData.Host` is always an empty string. Even worse `urlData.Host` may contain the port in it again.

Citing the documentation for type URL:
> Host        string    // host or host:port

Testing `scheme://host:port` results in `urlData.Host` being empty, parsing `scheme://host:port/` results in `urlData.Host` containing `host:port` as documented.

Reply:
Ok, thanks for checking 👍